### ISSUE TITLE
ldap_modify_batch(): Each element of a 'values' array must be a string

### ISSUE THREAD

**Issue by beeyev** [Adldap2/Adldap2#321]
In php 7, this function does not work as it supposed to.

**class BatchModification**
```
   public function setValues(array $values = [])
    {
        array_walk($values, function (&$value) {
            // We need to make sure all values given to a batch modification are
            // strings, otherwise we'll receive an LDAP exception when
            // we try to process the modification.
            $value = (string) $value;
        });

        $this->values = $values;

        return $this;
    }
```

It raises an error when you are going to save data:
> ldap_modify_batch(): Each element of a 'values' array must be a string

I don't know why, but this variant solves the problem

```
  public function setValues(array $values = [])
    {
        $values = array_map(function($value){
            return (string) $value;
        }, $values);

        $this->values = $values;
        return $this;
    }
```

**Comment by stevebauman** [comment]
Hi @beeyev.

That's strange... Probably something to due with referencing the value in a walk? Was this type of reference changed in PHP 7 do you know?

I'll test this quick and then put out a new release. Thanks for the report!

**Comment by beeyev** [comment]
Sorry Steve, i dont know. Unfortunately i can't test it on different php version right now.

**Comment by Naugrimm** [comment]
Encountered this today, too (PHP7). Can confirm that it does not occur under PHP5.6.

I think the problem comes from the following change:

PHP5.6: https://github.com/php/php-src/blob/PHP-5.6.25/ext/ldap/ldap.c#L1778
PHP7: https://github.com/php/php-src/blob/PHP-7.0/ext/ldap/ldap.c#L1821

Not sure if this is a bug in PHP itself. It is somewhat ... strange ... that something is not a string even if it is cast to a string explicitely.

**Comment by stevebauman** [comment]
Thanks for the research @Naugrimm! Fixed in latest release `v6.1.5`.
